src/main/java/org/sagebionetworks/web/client/widget/table/v2/schema/ColumnModelsWidget.java

```diff
@@ -105,11 +105,11 @@ public void configure(EntityBundle bundle, boolean isEditable) {
 		this.bundle = bundle;
 		List<ColumnModel> startingModels = bundle.getTableBundle().getColumnModels();
 		viewer.configure(ViewType.VIEWER, this.isEditable);
-		// Certain rows in datasets may not be editable because they reference locked versions.
-		boolean isEditableView = isEditable && (bundle.getEntity() instanceof EntityView || bundle.getEntity() instanceof SubmissionView);
+		// We can get the default columns/annotations for views and datasets
+		boolean isEditableViewOrDataset = isEditable && (bundle.getEntity() instanceof EntityView || bundle.getEntity() instanceof SubmissionView || bundle.getEntity() instanceof Dataset);
 		tableType = TableType.getTableType(bundle.getEntity());
-		editor.setAddDefaultViewColumnsButtonVisible(isEditableView);
-		editor.setAddAnnotationColumnsButtonVisible(isEditableView);
+		editor.setAddDefaultViewColumnsButtonVisible(isEditableViewOrDataset);
+		editor.setAddAnnotationColumnsButtonVisible(isEditableViewOrDataset);
```

Comment:
Addresses [SWC-5857](https://sagebionetworks.jira.com/browse/SWC-5857) -- buttons to add default columns or annotation columns were missing for datasets. I think in my first pass at this, I thought this was the query results editor, so the comment I left was irrelevant.